### PR TITLE
Split testing-base into separate infrastructure project

### PR DESCRIPTION
**Pull request by jvandort**
Previously :testing-base included both code that ran in the gradle daemon and in the test worker. This meant that test worker code could accidentally depend on code that it should not -- like things from :core. Previously, developers had to be sure to not accidentally depend on these restricted dependencies, and the build did not enforce that boundry.

This commit splits :testing-base into a separate :testing-base-infrastructure, where the infrastructure project contains all code that runs on workers and does not depend on projects that are not intended to run in workers.

### Reviewing cheatsheet

Before merging the PR, comments starting with 
- ❌ ❓**must** be fixed
- 🤔 💅 **should** be fixed
- 💭 **may** be fixed
- 🎉 celebrate happy things

### CODE REVIEW

**Comment by jvandort** [comment]
@bot-gradle test this

**Comment by jvandort** [comment]
@bot-gradle test this

File: platforms/software/testing-base/build.gradle.kts [hunk 1]
```diff
@@ -2,8 +2,6 @@ plugins {
     id("gradlebuild.distribution.api-java")
 }
 
-gradlebuildJava.usedInWorkers()
-
 description = """Basic testing related plugins, which establish conventions for testing output directories,
 and setup basic testing-related features lik a testSuites container and the testing extension.  It provides most of the
```
Comment (big-guy) [inline platforms/software/testing-base/build.gradle.kts:6]:
```suggestion
and setup basic testing-related features like a testSuites container and the testing extension.  It provides most of the
```

File: platforms/software/testing-base-infrastructure/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+dependencies {
+    api(projects.concurrent)
+    api(projects.javaLanguageExtensions)
+    api(projects.serialization)
+    api(projects.time)
+    api(project(":base-services"))
+    api(project(":logging"))
+    api(project(":logging-api"))
+    api(project(":worker-processes"))
```
Comment (big-guy) [inline platforms/software/testing-base-infrastructure/build.gradle.kts:17]: ❌ We should pick a format (project accessors or stringy paths) and make it consistent in a single file.

**Review by big-guy — APPROVED** [review]

**Pull request merged**